packages/vimeo/lib/pattern.js

```diff
@@ -7,11 +7,15 @@
  *
  * 1. Arbitrary whitespace
  * 2. Arbitrary whitespace
- * 3. The video ID. Numeric ID maxes out at 20 digits. Why 20? It's the
+ * 3. Vimeo domain, pathname, params, fragments.
+ * 4. The video ID. Numeric ID maxes out at 20 digits. Why 20? It's the
  * 		max length of a 64-bit integer, which Vimeo will be unlikely to ever
- * 		reach. Discards everything after the numeric ID.
- * 4. Arbitrary whitespace
- * 5. Arbitrary whitespace
+ * 		reach.
+ * 5. Privacy hash used by unlisted videos, if it's part of the pathname.
+ * 		(Can also be a param, so check for ?h=abc123 separately). Hashes are likely
+ * 		hex ([0-9a-f]), but hard to be sure, so allowing full alphanumeric just in case.
```

Comment:
Note to self: I suspect privacy hashes are always encoded as hex values, and probably a fixed length. However, I don't know for sure, so the pattern accepts up to 20 alphanumeric characters just in case.